src/beast/include/beast/impl/basic_streambuf.ipp

```diff
@@ -638,9 +641,9 @@ basic_streambuf<Allocator>::consume(size_type n)
             }
             else
             {
-                in_size_ -= avail;
-                if(out_pos_ != out_end_||
-                    out_ != list_.iterator_to(list_.back()))
+                in_size_ = 0;
+                if(out_ != list_.iterator_to(list_.back()) ||
+                    out_pos_ != out_end_)
```

Comment:
I'd add a comment to the `else` part of this if about "Both the input and output are empty."